python/nav/web/api/v1/views.py

```diff
@@ -1153,3 +1158,133 @@ class ModuleViewSet(NAVAPIMixin, viewsets.ReadOnlyModelViewSet):
+    If no mac address was supplied, an empty dict will be returned.
+
+    Example GET request: `/api/1/vendor/?mac=aa:bb:cc:dd:ee:ff`
+
+    Example GET response: `{"aa:bb:cc:dd:ee:ff": "Vendor A"}`
+
+    Example POST request:
+        `/api/1/vendor/` with body `["aa:bb:cc:dd:ee:ff", "11:22:33:44:55:66"]`
+
+    Example POST response:
+        `{"aa:bb:cc:dd:ee:ff": "Vendor A", "11:22:33:44:55:66": "Vendor B"}`
+    """
+
+    @staticmethod
+    def get(request):
+        mac = request.GET.get('mac', None)
+        if not mac:
+            return Response({})
+
+        try:
+            validated_mac = MacAddress(mac)
+        except ValueError:
+            return Response(
+                f"Invalid MAC address: '{mac}'", status=status.HTTP_400_BAD_REQUEST
+            )
+
+        results = get_vendor_names([validated_mac])
+        return Response(results)
+
+    @staticmethod
+    def post(request):
+        if isinstance(request.data, list):
+            mac_addresses = request.data
+
+        # This adds support for requests via the browseable API
+        elif isinstance(request.data, QueryDict):
+            json_string = request.data.get('_content')
+            if not json_string:
+                return Response("Empty JSON body", status=status.HTTP_400_BAD_REQUEST)
+            try:
+                mac_addresses = json.loads(json_string)
+            except json.JSONDecodeError:
+                return Response("Invalid JSON", status=status.HTTP_400_BAD_REQUEST)
+            if not isinstance(mac_addresses, list):
+                return Response(
+                    "Invalid request body. Must be a JSON array of MAC addresses",
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+        else:
+            return Response(
+                "Invalid request body. Must be a JSON array of MAC addresses",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            validated_mac_addresses = validate_mac_addresses(mac_addresses)
+        except ValueError as e:
+            return Response(str(e), status=status.HTTP_400_BAD_REQUEST)
+
+        results = get_vendor_names(validated_mac_addresses)
+        return Response(results)
+
+
+@django.db.transaction.atomic
+def get_vendor_names(mac_addresses: Sequence[MacAddress]) -> dict[str, str]:
+    """Get vendor names for a sequence of MAC addresses.
+
+    :param mac_addresses: Sequence of MAC addresses in a valid format
+        (e.g., "aa:bb:cc:dd:ee:ff").
+    :return: A dictionary mapping MAC addresses to vendor names. If the vendor for a
+        given MAC address is not found, it will be omitted from the response.
+    """
+    # Skip SQL query if no MAC addresses are provided
+    if not mac_addresses:
+        return {}
+
+    # Generate the VALUES part of the SQL query dynamically
+    values = ", ".join(f"('{str(mac)}'::macaddr)" for mac in mac_addresses)
```

Comment:
The use of `str` in the f-string seems redundant...

```suggestion
    values = ", ".join(f"('{mac}'::macaddr)" for mac in mac_addresses)
```